pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -184,42 +186,88 @@ func (c *defaultMaintenanceControl) Maintain(shootObj *gardencorev1beta1.Shoot,
 		handleError(fmt.Sprintf("Could not update the Shoot specification: %s", err.Error()))
 		return err
 	}
-	msg := "Completed; updated the Shoot specification successfully."
+
+	msg := "Maintenance complete. No action required."
+	if updatedKubernetesVersion != nil || updatedMachineImages != nil {
+		msg = "Maintenance complete."
+		if updatedMachineImages != nil {
+			msg = fmt.Sprintf("%s %s %s.", msg, "Updated ", *messageMachineImageUpdate)
+		}
+		if updatedKubernetesVersion != nil {
+			msg = fmt.Sprintf("%s %s %s.", msg, "Updated ", *messageKubernetesUpdate)
+		}
+	}
+
 	shootLogger.Infof("[SHOOT MAINTENANCE] %s", msg)
 	c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardencorev1beta1.ShootEventMaintenanceDone, "%s", msg)
 
 	return nil
 }
 
 // MaintainKubernetesVersion determines if a shoots kubernetes version has to be maintained and in case returns the target version
-func MaintainKubernetesVersion(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile) (*string, error) {
-	shouldBeUpdated, err := shouldKubernetesVersionBeUpdated(shoot, profile)
+func MaintainKubernetesVersion(shoot *gardencorev1beta1.Shoot, profile *gardencorev1beta1.CloudProfile, logger *logrus.Entry) (*string, *string, error) {
+	shouldBeUpdated, reason, isExpired, err := shouldKubernetesVersionBeUpdated(shoot, profile)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if shouldBeUpdated {
-		newerPatchVersionFound, latestPatchVersion, err := gardencorev1beta1helper.DetermineLatestKubernetesPatchVersion(profile, shoot.Spec.Kubernetes.Version)
+		// if on expired version 1.16.1 & there is preview version 1.16.2 -> do I then update minor version or do i go to preview version
```

Comment:
???

Reply:
Not deleted comment. Thanks for seeing it.